client/client.go

```diff
@@ -394,9 +394,35 @@ type RemoteContext struct {
 	// preference.
 	Platforms []string
 
-	// MaxConcurrentDownloads is the max concurrent content downloads for each pull.
+	// MaxConcurrentDownloads restricts the total number of concurrent downloads
+	// across all layers during an image pull operation. This helps control the
+	// overall network bandwidth usage.
 	MaxConcurrentDownloads int
 
+	// MaxConcurrentDownloadOperations limits the number of parallel operations
+	// during an image pull to prevent system resource exhaustion.
+	//
+	// These operations include:
+	// - Downloading a layer
+	// - Unpacking a layer into the snapshotter
+	// - Moving an unpacked layer to its final location
+	MaxConcurrentDownloadOperations int
+
+	// MaxConcurrentDownloadsPerLayer controls parallel downloading of image layers
+	// by splitting them into chunks.
+	//
+	// - Values <= 1: Layers are downloaded using a single connection (default).
+	// - Values > 1: Layers are divided into chunks and downloaded in parallel,
+	//   which can significantly reduce pull times for large layers.
+	MaxConcurrentDownloadsPerLayer int
```

Comment:
also on a side note: In the current PR we add this in 4 places: ctr, go client, cri, and transfer service.

Given it's a new feature and transfer service is used by default in containerd 2.0, I wonder if we should just add this only in transfer service. cc @dmcgowan

Reply:
Adding this everywhere makes it easier to test things, using `ctr` & `cri`.

Also, I can see a scenario where someone wants to manually test the speed with their parameters depending on machine, etc.
And having it in `ctr`/`cri` kinda makes sense.
At which point I was like, yup adding it everywhere for consistency.

Reply:
For CRI specifically, I think the goal is to actually migrate existing config to transfer service, e.g.,

https://github.com/containerd/containerd/blob/d30521491d87d8172a4c922fff35e4afa660e63e/internal/cri/config/config.go#L311-L313

Reply:
Oh, I see ! I think that to test and all, we should keep these, and the migration PR should move them.
But if then migration PR happens first, then I'll adapt the code. WDYT ?

Reply:
SGTM, fyi the migration PR is this one https://github.com/containerd/containerd/pull/8515